tiatoolbox/__init__.py

```diff
@@ -101,14 +106,16 @@ def read_registry_files(path_to_registry: str | Path) -> dict | str:
 rcParam["pretrained_model_info"] = read_registry_files("data/pretrained_model.yaml")
 
 
-def _lazy_import(name: str, module_location: Path) -> sys.modules:
+def _lazy_import(name: str, module_location: Path) -> dict[str, ModuleType]:
     spec = importlib.util.spec_from_file_location(name, module_location)
+    if spec is None or spec.loader is None:
+        raise ModuleNotFoundError(name=name, path=str(module_location))
     loader = importlib.util.LazyLoader(spec.loader)
     spec.loader = loader
     module = importlib.util.module_from_spec(spec)
     sys.modules[name] = module
     loader.exec_module(module)
-    return module
+    return {name: module}
```

Comment:
What's the benefit of returning this as a dictionary?

Reply:
The return type for this function used to be `sys.modules`, but `sys.modules` is a variable so we can't use it as type. I looked up the type of `sys.modules`, it was `dict[str, ModuleType]` so I changed the return type to this, don't think there is any benefit of returning a dictionary.

Alternatively we change the return type of this function to `ModuleType` and keep everything else the same.